api/v1alpha1/envoyproxy_types.go

```diff
@@ -264,6 +264,13 @@ type EnvoyProxyKubernetesProvider struct {
 	//
 	// +optional
 	EnvoyHpa *KubernetesHorizontalPodAutoscalerSpec `json:"envoyHpa,omitempty"`
+
+	// DisablePortShift disables the port shifting feature in the Envoy Proxy.
+	// When set to false (default value), if the service port is a privileged port (1-1023), add a constant to the value converting it into an ephemeral port.
+	// This allows the container to bind to the port without needing a CAP_NET_BIND_SERVICE capability.
+	//
+	// +optional
+	DisablePortShift *bool `json:"disablePortShift,omitempty"`
```

Comment:
any other alternate name suggestions ?`DisablePortShift` is too vague imo and doesnt account for the fact that this issue is tied to `ContainerPort` 
cc @envoyproxy/gateway-maintainers @envoyproxy/gateway-reviewers

Reply:
To come up with variable names is harder than writing code.

Something like `DisableShiftContainerPort`?

Reply:
Could it be an other way around, like BindPort or StaticPort bool value?

Reply:
`BindPort` and `StaticPort` looks like fields for port value. Not for bool

Reply:
I prefer `UseListenerPortAsContainerPort` from https://github.com/envoyproxy/gateway/pull/2405/files
@envoyproxy/gateway-maintainers / @envoyproxy/gateway-reviewers wdyt

Reply:
+1 for `UseListenerPortAsContainerPort` .

Reaons: I can tell what `UseListenerPortAsContainerPort` does by its name, but can't infer what `DisablePortShift`  without looking into the comment.

Reply:
No problem. I can use this variable. However, I see a small logical issue. If I plan to use a port higher than 1024 in the Listener (for example, 3000), there will be no 'port offsets'. That means the `UseListenerPortAsContainerPort` variable will be set to `false` (the default value), and at the same time, the listener port and the container port will be equal.